### DISCUSSION TITLE
Is there a way to toggle stellar aberration off?

### DISCUSSION THREAD

**Discussion by pj-camp** [Stellarium/stellarium#3378] [Q&A] [answered]
I need to do this as part of a parallax activity with non-science majors. Thanks!

**Comment by gzotti** [comment, marked as answer]
Figure 4.7 in the User Guide shows the relevant checkbox.

**Comment by pj-camp** [comment]
Thanks! Sometimes systems are so complicated that you can't see the trees
for the forest.

FWIW, if you do a Google search for that, the top hit is a seriously out of
date one from elsewhere.

On Mon, 28 Aug 2023 at 09:33, Georg Zotti ***@***.***> wrote:

> Figure 4.7 in the User Guide shows the relevant checkbox.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stellarium/stellarium/discussions/3378#discussioncomment-6842641>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJSMTKADJYRQRGL6JBLXXSM37ANCNFSM6AAAAAA4BNPJB4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose

**Reply by 10110111** [reply]
What exactly did your search query look like?

**Comment by pj-camp** [comment]
stellarium user guide

On Mon, 28 Aug 2023 at 11:56, Ruslan Kabatsayev ***@***.***>
wrote:

> What exactly did your search query look like?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stellarium/stellarium/discussions/3378#discussioncomment-6844161>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJSXLGWWQEHFQNZ6YALXXS5TBANCNFSM6AAAAAA4BNPJB4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose

**Reply by 10110111** [reply]
I get a 23.1 User Guide at `uny.ac.id` as the first result. Not "seriously" out of date, but yes, not the latest one. A bigger question is why Stellarium's website only appears in a second search result—and only as the main page of the project, rather than the User Guide.
I guess the problem is that `stellarium.org` only provides a _download_ of the user guide file—rather than a version readable in the browser. @alex-w @gzotti The link may need some change as discussed e.g. [here](https://webapps.stackexchange.com/q/48061).

**Reply by gzotti** [reply]
The User Guide for 0.12/0.13 or so resided in a wiki, online. However, a searchable PDF which is developed in parallel to the application by the developers is much better than a wiki which was developed by non-developers. This is why the User Guide PDF comes installed onto users' harddisks/SSDs and linked in the start menu on the majority platform. No need to google around for 3rd-party information which may be outdated or wrong. The aim of providing an online HTML version (without searchable index?) for those without a PDF reader was stuck in development because ... well. ... I just don't need it, and many other things were more important. 

I totally hate it when I click on a download link to get a large PDF and it opens in my browser. I want to store it immediately where I want to keep it, and may not even read it immediately. 

I don't know why somebody in India or elsewhere hosts the Stellarium User Guide. This only prevent us from getting reliable download counts.

**Reply by 10110111** [reply]
> I totally hate it when I click on a download link to get a large PDF and it opens in my browser.

Well, this is how the major sources of PDFs behave: arXiv, for-profit scientific publishers, etc. And, this behavior is easy to override by *RMB* → *Save link as...* → <kbd>Return</kbd>, unlike the opposite behavior when you have to go and remove a file that you just wanted to look through and close.
Anyway, if you feel so strongly about this, then you could provide two links: view, download. This may be able to let the search engines read the viewable version and index it, hopefully getting it to the top of the search results.

> the User Guide PDF comes installed onto users' harddisks/SSDs and linked in the start menu <...> No need to google around

This is just not how the users work nowadays. I, for one, would google for the user guide if I didn't *explicitly* know it's already there in the installation directory. And the Start Menu doesn't help discover it either: who even uses the menu when all the launchers are already on the desktop?
Moreover, I often google for e.g. Qt docs instead of opening Qt Assistant despite having it installed—simply because Google often uses better matching algorithms for search queries, rather than the verbatim search available in the viewer. So it's better to have the Guide available for reading online — not necessarily as HTML, but at least as a viewable PDF.

**Reply by gzotti** [reply]
Yes, I know (and suffer from Academia). The button says "Download PDF". It does not say "Open PDF". 

Provide two links: This will only confuse those who so far had difficulties finding the User Guide on their harddisk. The proper solution would be completion of the htlatex/tex4ht issues (See guide/Makefile) and putting the result online. Else people will download 30MB guide into their browser every time they need to find a button.

Launchers of the desktop: Yes, 4k desktops cluttered with 300+ icons are another modern disgrace. Do you find anything? I always uncheck "add icon to desktop". The favorites are attached to the task bar or start menu. The rest is in the structured menu, although this gets worse in every release of Windows since 7. If desktop icons are favoured by the majority, then why not add the icon for the User Guide to the installer? (@alex-w ?)

Not sure about Qt Assistant, never opened a stand-alone program of that name. I use the built-in help from QtCreator, and occasionally the online version (to check for version differences). The point is, the installed SUG is more adequate than anything google could come up with.

**Reply by 10110111** [reply]
> Yes, 4k desktops cluttered with 300+ icons are another modern disgrace. Do you find anything? I always uncheck "add icon to desktop".

Well, whenever I have to use Windows (which is rare), I just move the icons I need to a definite place and remove all the others.

> If desktop icons are favoured by the majority, then why not add the icon for the User Guide to the installer?

IMO the existing set is already ugly enough, since ideally software should just work rather than require kludgy "modes". Adding one more icon would just add to the chaos. Maybe we should add an easy to locate button to open the User Guide from the _Help_ dialog. It would provide a function similar to the typical _Help_ → _User Manual_ menu.

**Reply by alex-w** [reply]
Technically, we can add another one icon on the desktop, but I don't like it!

**Reply by alex-w** [reply]
> I guess the problem is that stellarium.org only provides a download of the user guide file—rather than a version readable in the browser. The link may need some change as discussed e.g. [here](https://webapps.stackexchange.com/q/48061).

The SUG is too big for viewing via Google PDF Viewer, so, the solution will not work. We may put the SUG into repository for stellarium.org website, but I fear adding big binary files into git repository not very good idea.

**Reply by 10110111** [reply]
OK, but what about my suggestion to add a button to the _Help_ dialog?

**Reply by alex-w** [reply]
> OK, but what about my suggestion to add a button to the Help dialog?

I fear the "Further Reading" section nobody reads, as and first page of stellarium.org website, but yes, we can add link to the latest version of SUG.

Adding new tab "User Guide" are pretend to add the SUG as HTML pages into main package - enlargement the installer at least on 40 MB. Of course in this case we should remove the guide.pdf file from the installation package (~30MB).

**Reply by 10110111** [reply]
No, I mean add a button "Open User Guide", which would just let the OS open the locally-installed PDF file in an external viewer.

**Reply by gzotti** [reply]
Yes, that may be a good idea. At least on Windows, when we know where the Guide is installed (and where most users report issues). I don't know, is the Guide packed with the other distributions?
The button should carry some PDF icon to let users better understand (and to stand out by its red signal color).

**Reply by 10110111** [reply]
> I don't know, is the Guide packed with the other distributions?

Ubuntu doesn't appear to include it: neither in `stellarium`, nor in `stellarium-data`. This might be because the CMake build process doesn't integrate generation of the PDF. Is the PDF included in the release source tarballs?

**Reply by alex-w** [reply]
> No, I mean add a button "Open User Guide", which would just let the OS open the locally-installed PDF file in an external viewer.

Currently we add the guide.pdf file into Windows-package only. I guess I can add it into macOS package too, but for linux/BSD it can be problematic.

**Reply by 10110111** [reply]
> for linux/BSD it can be problematic

Why?

**Reply by alex-w** [reply]
> Ubuntu doesn't appear to include it: neither in stellarium, nor in stellarium-data. This might be because the CMake build process doesn't integrate generation of the PDF. Is the PDF included in the release source tarballs?

No, we include TeX source into tarball, but not final PDF. Of course in shoud be `stellarium-guide` package.

**Reply by alex-w** [reply]
> Why?

A few years ago, we're start prepare sources for maintainers to producing PDF file of SUG, but I didn't see guide in linux distros.

**Reply by gzotti** [reply]
Creation of the PDF (with all temporary files) runs in the source directory. It may be cleaner, but may be a challenge to set up properly, to also build it in cmake's build directory for full-blown releases. However I'd prefer to keep it running also as it is now for practical reasons. "Make pdf" runs several times, with some helper programs, while for some minor edits, I like to run pdflatex manually for single runs. 

Addition of a stellarium-guide package for Linux (as "recommended" package) sounds like a good idea.

Now, that new button could use some OS-own "open guide.pdf", but should be prepared to handle "file not found" with an instruction dialog to "please install stellarium-guide package" or "please download guide and place into...  so that Stellarium can find it".

**Reply by 10110111** [reply]
> Addition of a stellarium-guide package for Linux (as "recommended" package) sounds like a good idea.

Most Linux (and BSD?) users use packages from their repositories, so there must be a way for packagers to actually learn about the off-build-system build of the Guide. I suppose there's no `stellarium-guide` package in e.g. Ubuntu simply because the packagers don't know about this.

**Reply by gzotti** [reply]
Yes, fully agreed. Some cmake master must find out how to work with LaTeX files in another directory. Maybe this exists with an easy switch, I have not dived into this area yet.

**Reply by 10110111** [reply]
BTW, I just noticed that when I enter "stellarium user guide" into Google search entry field, it suggests "stellarium user guide pdf", which means users do want to find the user guide in PDF form online.

The main problem with discoverability of the latest version seems to be the following (the "learn why" link points to [this page](https://support.google.com/webmasters/answer/7489871?hl=en)):

![Screenshot_2023-08-29_23-34-35](https://github.com/Stellarium/stellarium/assets/6376882/fb46fb66-b0f2-47dc-b2c9-9845f9d6ccde)

**Reply by gzotti** [reply]
Good find. However, I don't even see a robots.txt in our website's git.  The verification requires some extra files from google on our website. @alex-w Shall I add that to the website git and try to confirm ownership? Not sure what happens if there not even is a robots.txt...

**Reply by 10110111** [reply]
The relevant site is `github.com`. I think we need to somehow pass a link to a website we can actually control, since GitHub doesn't seem to have any means to control its `robots.txt`.

**Reply by gzotti** [reply]
I feared that. OK, but there must be a way for the many github users to unblock their webpages.

**Reply by 10110111** [reply]
But it's not "their webpage": it's a downloadable asset from a release. The rules for search engines' crawling of these assets are defined by GitHub rather than the projects hosted there.

**Reply by gzotti** [reply]
New version:
OK, but there must be a way for the many github users to solve such a problem.

**Reply by 10110111** [reply]
Well, at least none of the search terms I tried yielded any relevant results. This seems to be an uncommon problem. I don't think we can control the behavior of the GitHub UI with respect to search engines. Better host the PDF at `stellarium.org` and provide all the necessary adjustments there.

**Reply by alex-w** [reply]
Please check https://stellarium.org/guide/

**Reply by 10110111** [reply]
This looks awful:

1. A veeeery long HTML page with ToC hyperlinks pointing to different parts of it. This is hard to use. I'd rather have different chapters and subchapters on their own separate pages, with all the cross-references pointing to them.
2. Text selection feels sluggish, at first I even though it doesn't work. Plus the color... why pale cyan?! Why not simply use the browser's default?
3. The page forces smooth scrolling on me. I can disable Javascript to avoid this, and thankfully, it doesn't break the ability to display the content. But still, I explicitly disabled smooth scrolling in the browser settings, why would a web page force it on or emulate it?
1. Grossly inefficient, see e.g. `<div class="t m0 x4 h6 y214 ff1 fs4 fc0 sc0 ls0 ws0">E.2<span class="_ _c1"> </span>Planetar<span class="_ _0"></span>y<span class="_"> </span>Coordinate<span class="_"> </span>Systems<span class="_ _c3"> </span>388</div>` for the title of E.2. So loading time is much longer than it would take with a PDF.
4. RMB click on the background white results in a context menu as if I clicked an image ("Save image as..." etc.), which seems to imply that these white "pages" are images — not sure how this impacts performance and use of bandwidth.

**Comment by pj-camp** [comment]
Who can fathom the methods of Google? The top hit on my search is for
version 0.10.2 at http://www.astronomia.edu.uy.

On Mon, 28 Aug 2023 at 12:14, Ruslan Kabatsayev ***@***.***>
wrote:

> I get a 23.1 User Guide at uny.ac.id as the first result. Not "seriously"
> out of date, but yes, not the latest one. A bigger question is why
> Stellarium's website only appears in a second search result—and only as the
> main page of the project, rather than the User Guide.
> I guess the problem is that stellarium.org only provides a *download* of
> the user guide file—rather than a version readable in the browser. @alex-w
> <https://github.com/alex-w> @gzotti <https://github.com/gzotti> The link
> may need some change as discussed e.g. here
> <https://webapps.stackexchange.com/q/48061>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stellarium/stellarium/discussions/3378#discussioncomment-6844382>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJUMWHAK62XSUS54Q5DXXS7VFANCNFSM6AAAAAA4BNPJB4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose

**Reply by gzotti** [reply]
Hmm, one of those sad long-dead websites. Last update 2009? The webmaster's own site is no longer accessible, so probably his email also goes nowhere. As long as that university does not review their websites, this will just continue. 
If I search for Stellarium, I would prefer to click on stellarium.org, and not things like astro.edu.ie/resources/stellarium or stellarium.download-free-software.com. But we cannot prevent occurrences of the latter.

**Comment by pj-camp** [comment]
Probably for local student access. The one I found was apparently in
Uruguay where I imagine Internet access is not all that reliable. I just
wonder how it floated to the top of a Google search.

On Mon, 28 Aug 2023 at 16:12, Georg Zotti ***@***.***> wrote:

>
> I don't know why somebody in India or elsewhere hosts the Stellarium User
> Guide. This only prevent us from getting reliable download counts.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stellarium/stellarium/discussions/3378#discussioncomment-6846299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJWVOIDY3C5L5TVBWELXXT3RLANCNFSM6AAAAAA4BNPJB4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose

**Reply by gzotti** [reply]
Yes, I'm sure about that. Just that the responsible person vanished, apparently in or short after 2009. And yes, what did he do to keep floating?

**Reply by 10110111** [reply]
> what did he do to keep floating?

Oh, I'm sure he did nothing for this. It's the title of the document that did it: it's equal to the search term: "Stellarium User Guide". Compare to the less outdated one: "Stellarium 23.1 User Guide".
